app/src/main/java/com/markkolenbrander/capstonenewsapp/MainActivity.kt

```diff
@@ -0,0 +1,136 @@
+package com.markkolenbrander.capstonenewsapp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+import androidx.core.view.children
+import com.markkolenbrander.capstonenewsapp.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private val articlesCollection = mutableMapOf<String, MutableList<String?>>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        addArticles()
+        makingTextViews(articlesCollection)
+    }
+
+    private fun addArticles(){
+        articlesCollection["author"] = mutableListOf(
+            "n-tv NACHRICHTEN-1", "finanzen.net Redaktion", null , "Michael Bassewitz", "Sascha Mattke"
+        )
+        articlesCollection["title"] = mutableListOf(
+            "14:00 Aktienkurs von Softwarekonzern geht durch die Decke - n-tv NACHRICHTEN",
+            "US-Inflation hat Höhepunkt erreicht: Jim Cramer sieht das Paradies für Aktien kommen - finanzen.net",
+            "Theion: Fragwürdige Wunderakkus aus Berlin - Golem.de - Golem.de",
+            "„Weich-Währung“: Experten schlagen wegen Euro-Absturz Alarm - BILD",
+            "Tesla Model Y jetzt als Basis-Version für Europa bestellbar, Produktion in China (aktualisiert) - TeslaMag.de"
+        )
+        articlesCollection["sourceName"] = mutableListOf(
+            "ABC News",
+            "Ars Technica",
+            "Axios",
+            "Australian Financial Review",
+            "Bloomberg"
+        )
+    }
+
+    private fun makingTextViews(articles: MutableMap<String, MutableList<String?>>){
+
+        val mainGroup = binding.llMain
+        var count = 0
+        for (text in mainGroup.children){
+            if (text is TextView){
+
+                val content = (
+                        if (articles["author"]?.get(count) != null) {
+                            articles["author"]?.get(count) + "\n"
+                        }else {"...UNKNOWN..." + "\n"} +
+                                articles["title"]?.get(count) + "\n" +
+                                articles["sourceName"]?.get(count)
+                        )
+                text.text = content
+                count ++
+            }
+        }
+    }
+
+    // I leave this hardcoded List's just in case we have to use them later on
+    private val source = arrayListOf(
```

Comment:
Good call to leave these here, as we move forward through the next couple weeks we'll reference the usage of these data class models a bit more so it's good to have this data around.

Reply:
Thanks